lib/Server.js

```diff
@@ -2011,15 +2011,15 @@ class Server {
       stats &&
       (!stats.errors || stats.errors.length === 0) &&
       (!stats.warnings || stats.warnings.length === 0) &&
-      stats.assets &&
-      stats.assets.every((asset) => !asset.emitted);
+      this.currentHash === stats.hash;
```

Comment:
@markjm I remove unnecessary `??` check, because it is redundant, when client connected to dev server we always send stats (so we write currentHash), on initial run we send stats too and write hash (client can connected, but compilation can be still not finished, so after it will be finished we should always send stats, it will be never `still-ok`)

Reply:
Also don't forget about disconnected, developer can run dev server again and compilation can take time (i.e. client connected, but no stats right now) and we should send stats always too